dbus-cpp: init at 5.0.3

OPNA2608 opened the pull request:
## Description of changes

Working towards #99090.

[dbus-cpp](https://gitlab.com/ubports/development/core/lib-cpp/dbus-cpp), a C++11 dbus-binding. Repo says it's header only, *squints at monitor* but it builds a library to link against. Oh well. Needed by Lomiri's location service, biometry daemon, mediascanner2 service & trust store.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

OPNA2608 committed:
dbus-cpp: init at 5.0.3

OPNA2608 committed:
dbus-cpp: Disable tests

Tests seem very flaky on some systems / for some ppl. Consistently fail on OfBorg ARM CI, past reports of
sporadic test failures on amd64. Not sure if due to weak systems or DBus weirdness. Disabling for now.

mrcjkb commented:
Result of `nixpkgs-review pr 260941` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>dbus-cpp</li>
    <li>dbus-cpp.dev</li>
    <li>dbus-cpp.doc</li>
    <li>dbus-cpp.examples</li>
  </ul>
</details>

OPNA2608 committed:
dbus-cpp: Fix pkg-config file

lomiri.mediascanner2 errored due to includedir being a non-existent path.
TIL our pkg-config patcher requires ${prefix} to work.

OPNA2608 commented:
Re-tested by packaging more packages that depend on this, and eventually noticed that it was missing a fix:

```
CMake Error in src/qml/MediaScanner.0.1/CMakeLists.txt:
  Imported target "PkgConfig::DBUSCPP" includes non-existent path

    "/nix/store/jz9cc5mrw7vw9mrqmyp4xs1rhn0q4p4v-dbus-cpp-5.0.3/include"
```

Should be good-er now.

mrcjkb commented:
Result of `nixpkgs-review pr 260941` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>dbus-cpp</li>
    <li>dbus-cpp.dev</li>
    <li>dbus-cpp.doc</li>
    <li>dbus-cpp.examples</li>
  </ul>
</details>
